crates/interpreter/src/instructions/host.rs

```diff
@@ -413,13 +422,17 @@ pub fn auth<H: Host, SPEC: Spec>(interpreter: &mut Interpreter<'_>, host: &mut H
             .as_ref(),
     );
     message[33..65].copy_from_slice(interpreter.contract().address.into_word().as_ref());
-    message[65..97].copy_from_slice(commit.unwrap_or_default().as_ref());
+    message[65..97].copy_from_slice(commit.as_ref());
     let message_hash = revm_primitives::keccak256(&message);
 
     // Verify the signature
     let recovered = revm_primitives::secp256k1::ecrecover(&signature, &message_hash);
 
     if recovered.is_err() || Address::from_word(recovered.unwrap_or_default()) != authority {
+        // The spec states that the `AUTH` opcode returns 0 and unsets the authorized account
+        // if the signature is either invalid, or the recovered address does not match the
+        // authority.
+        interpreter.authorized = None;
```

Comment:
super super nit, but the order of these operations is switched between the success and failure cases, which makes it ever so slightly less readable